### PR TITLE
FAQ page uses virtual DOM APIs

### LINKED ISSUE #746: Make FAQ not hide the answers
From https://github.com/jupyter/jupyterlab/issues/325#issuecomment-232096794

> Think about maybe not hiding the response in a collapsible area, and just make a TOC a the top. This woudl allow users to search with browser Ctrl-F.

CC @charnpreetsingh185.

### PR DESCRIPTION
**Pull request by charnpreetsingh**
- jQuery removed from FAQ page
- Table of Contents at the top of the page
- Page scrolls down to selected question and highlights it briefly

Closes #746 

cc @spoorthyv

### CODE REVIEW

**Comment by ellisonbg** [comment]
cool i will have a look!

**Comment by afshin** [comment]
Looks good to me, the only question I have is for @ellisonbg about whether you think that the colors in this stylesheet should be private variables declared at the top of the file or if they are fine being used inline the way they are right now.

**Comment by afshin** [comment]
> jQuery removed from FAQ page

🏆

**Comment by ellisonbg** [comment]
To help simplify the patterns of working with the virtual DOM APIS, we have created a new `VDomModel` and `VDomWidget` classes that encode the best practices for working with this API. These new APIs are documented here:

https://github.com/jupyter/jupyterlab/blob/master/tutorial/virtualdom.md

Could you update this PR to use these classes? Please let me know if you have any questions!

**Comment by charnpreetsingh185** [comment]
@ellisonbg Yes, I saw that PR I will update this ASAP.

**Comment by ellisonbg** [comment]
@charnpreetsingh185 any progress on this one?

**Comment by charnpreetsingh185** [comment]
@ellisonbg I understand the use of this pattern for dynamic content but don't understand the use of it in the FAQ page. Could you explain how it could be used in this case?

**Comment by ellisonbg** [comment]
For the case of more static content, using the virtual DOM class just forces good software engineering design:

* It forces you to separate the model (information about each FAQ) from the view (the virtual DOM).
* It takes care of the boilerplate code for getting content onto the page.
* Things that you didn't think were dynamic, may actually be - for example, if you want to hide/collapse questions, then there would be something in the model for that
* Uses patterns that are common/familar wrt to the rest of the code base

**Comment by charnpreetsingh185** [comment]
@ellisonbg This is ready for another review.

File: src/faq/plugin.ts [hunk 1]
```diff
@@ -31,301 +29,330 @@ const faqExtension: JupyterLabPlugin<void> = {
...
+    );
+
+    // Create a section element that all other FAQ Content will go under.
+    let questionAnswerList =
+    h.section({className: CONTENT_CLASS},
+      h.h2({className: SUBHEADER_CLASS}, subHeadings[0]),
+      // Create list of questions/answers under the Basics section.
+      h.ul({className: FAQ_LIST_CLASS},
+        h.li({className: QUESTION_CLASS, id: 'basicsQ1'}, basicsQuestions[0]),
+        h.li({className: ANSWER_CLASS},
```
Comment (blink1073) [inline src/faq/plugin.ts]: Why not store the answers in the model as well?

Comment (charnpreetsingh185) [inline src/faq/plugin.ts]: @blink1073 The issue with storing answers in the model is that some answers have other DOM elements on them that makes it hard to represent simply as `strings`

File: src/faq/plugin.ts [hunk 2]
```diff
@@ -31,301 +29,330 @@ const faqExtension: JupyterLabPlugin<void> = {
...
+ */
+const ANSWER_CLASS = 'jp-FAQ-answer';
+
+/**
+ * The class name added to anchor elements.
+ */
+const ANCHOR_CLASS = 'jp-FAQ-a';
+
+
+class FaqModel extends VDomModel {
```
Comment (blink1073) [inline src/faq/plugin.ts]: @charnpreetsingh, I think with some added docstrings and making the attributes public like the previous PR this should be ready.

**Comment by charnpreetsingh** [comment]
@blink1073 changes made!

**Comment by blink1073** [comment]
@charnpreetsingh, would you also do us the honor of removing jquery from the `package.json` dependencies as well?  I verified that removing it has no ill effect along with this PR.

**Comment by charnpreetsingh** [comment]
@blink1073 changes made!

**Comment by blink1073** [comment]
Nicely done, @charnpreetsingh, clean sweep today!

**Pull request merged**